File: hanoi.scala
// This is Scala code
object Main {
	def main(args: Array[String]):Unit = hanoi(readLine().toInt,"a","b","c")
	def hanoi(n:Int, x:String, y:String, z:String):Unit = {
		if(n>1) hanoi(n-1,x,z,y)
		println(s"$x=>$y")
		if(n>1) hanoi(n-1,z,y,x)
	}
}

File: hanoi.erl
% This is Erlang code

-module(prog).
-export([main/0]).

main() ->
    hanoi(3,"a","b","c").

hanoi(1,X,_,Z) -> io:format("~s=>~s~n", [X, Z]);
hanoi(N,X,Y,Z) when N > 0 ->
    hanoi(N - 1, X, Z, Y),
    io:format("~s=>~s~n", [X, Z]),
    hanoi(N - 1, Y, X, Z).
